### ISSUE TITLE
client.views_open fails w/ internal_error

### ISSUE THREAD

**Issue by zachmetcalf** [slackapi/bolt-python#992]
When I register to the below shortcut which opens up a pretty trivial modal, it fails with an internal error.

### Reproducible in:

```python
@SlackApp.shortcut("shortcut_demo")
def shortcut_demo(ack, client, shortcut):
    ack()
    client.views_open(
        trigger_id=shortcut["trigger_id"],
        view={
            "type": "modal",
            "title": {
                "type": "plain_text",
                "text": "Title"
            },
            "submit": {
                "type": "plain_text",
                "text": "Submit"
            },
            "blocks": [
                {
                    "type": "section",
                    "text": {
                        "type": "mrkdwn",
                        "text": "Hello World"
                    }
                }
            ]
        }
    )
```

#### The `slack_bolt` version

latest (it's in my requirements.txt as slack-bolt)

#### Python runtime version

Python 3.11.4

### Expected result:

This should just open up the simple modal

### Actual result:

```
ERROR:slack_bolt.App:Failed to run listener function (error: The request to the Slack API failed. (url: https://www.slack.com/api/views.open)
The server responded with: {'ok': False, 'error': 'internal_error'})
Traceback (most recent call last):
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_bolt\listener\thread_runner.py", line 120, in run_ack_function_asynchronously
    listener.run_ack_function(request=request, response=response)
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_bolt\listener\custom_listener.py", line 50, in run_ack_function
    return self.ack_function(
           ^^^^^^^^^^^^^^^^^^
  File "T:\git\slackbot\slackbot\shortcuts\shortcut_demo.py", line 12, in shortcut_demo
    client.views_open(
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\client.py", line 4864, in views_open
    return self.api_call("views.open", json=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\base_client.py", line 155, in api_call
    return self._sync_send(api_url=api_url, req_args=req_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\base_client.py", line 186, in _sync_send
    return self._urllib_api_call(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\base_client.py", line 317, in _urllib_api_call
    ).validate()
      ^^^^^^^^^^
  File "t:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\slack_response.py", line 199, in validate
    raise e.SlackApiError(message=msg, response=self)
slack_sdk.errors.SlackApiError: The request to the Slack API failed. (url: https://www.slack.com/api/views.open)
The server responded with: {'ok': False, 'error': 'internal_error'}
```

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by mwbrooks** [comment]
Hi @zachmetcalf 👋🏻 Thanks for trying out Bolt for Python!

I used your code sample in our getting started app using the latest `slack_bolt` and it worked alright. Below is the code sample:

```python
import os
from slack_bolt import App
from slack_bolt.adapter.socket_mode import SocketModeHandler


# Initializes your app with your bot token and signing secret
app = App(token=os.environ.get("SLACK_BOT_TOKEN"))


@app.shortcut("open_modal")
def shortcut_demo(ack, client, shortcut):
    ack()
    client.views_open(
        trigger_id=shortcut["trigger_id"],
        view={
            "type": "modal",
            "title": {
                "type": "plain_text",
                "text": "Title"
            },
            "submit": {
                "type": "plain_text",
                "text": "Submit"
            },
            "blocks": [
                {
                    "type": "section",
                    "text": {
                        "type": "mrkdwn",
                        "text": "Hello World"
                    }
                }
            ]
        }
    )


# Start your app
if __name__ == "__main__":
    SocketModeHandler(app, os.environ["SLACK_APP_TOKEN"]).start()
```

1️⃣ Would you mind trying it again, in case the `internal_error` was a temporary Slack API issue?

2️⃣ If you still have the error, can you try my sample code above? You'll need to have `SLACK_BOT_TOKEN` and `SLACK_APP_TOKEN` setup and replace `open_modal` with `shortcut_demo`.

Thanks!

**Comment by zachmetcalf** [comment]
thanks for reaching out @mwbrooks! unfortunately, this doesn't work and I can't quite figure it out. I've simplified my code and setup the following example:

```
from dotenv import load_dotenv
from slack_bolt import App
from slack_bolt.adapter.socket_mode import SocketModeHandler
import json
import os

config = json.load(open("config/config.json"))

load_dotenv(config["env"])
SLACK_APP_TOKEN = os.getenv("SLACK_APP_TOKEN")
SLACK_BOT_TOKEN = os.getenv("SLACK_BOT_TOKEN")

# Initializes your app with your bot token and signing secret
app = App(token=SLACK_BOT_TOKEN)

@app.event("app_mention")
def event_app_mention(client, event):
    client.reactions_add(
        channel=event["channel"],
        timestamp=event["ts"],
        name="wave",
    )

@app.shortcut("open_modal")
def open_modal(ack, client, shortcut):
    ack()

    # post message works...
    client.chat_postMessage(
        channel="mychannel",
        text="hello world"
    )

    # views_open doesn't...
    client.views_open(
        trigger_id=shortcut["trigger_id"],
        view={
            "type": "modal",
            "title": {
                "type": "plain_text",
                "text": "Title"
            },
            "submit": {
                "type": "plain_text",
                "text": "Submit"
            },
            "blocks": [
                {
                    "type": "section",
                    "text": {
                        "type": "mrkdwn",
                        "text": "Hello World"
                    }
                }
            ]
        }
    )

if __name__ == "__main__":
    SocketModeHandler(app, SLACK_APP_TOKEN).start()
```

If I do this, the reaction event works great, the event does get triggered (and successfully posts the message). But it's just something about opening the view modal that fails. And it's the same error, see below:

Failed to run listener function (error: The request to the Slack API failed. (url: https://www.slack.com/api/views.open)
The server responded with: {'ok': False, 'error': 'internal_error'})
Traceback (most recent call last):
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_bolt\listener\thread_runner.py", line 120, in run_ack_function_asynchronously
    listener.run_ack_function(request=request, response=response)
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_bolt\listener\custom_listener.py", line 50, in run_ack_function
    return self.ack_function(
           ^^^^^^^^^^^^^^^^^^
  File "C:\git\slackbot\slackbot\__main__.py", line 41, in open_modal
    client.views_open(
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\client.py", line 4949, in views_open
    return self.api_call("views.open", json=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\base_client.py", line 155, in api_call
    return self._sync_send(api_url=api_url, req_args=req_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\base_client.py", line 186, in _sync_send
    return self._urllib_api_call(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\base_client.py", line 317, in _urllib_api_call       
    ).validate()
      ^^^^^^^^^^
  File "c:\git\slackbot\.venv\Lib\site-packages\slack_sdk\web\slack_response.py", line 199, in validate
    raise e.SlackApiError(message=msg, response=self)
slack_sdk.errors.SlackApiError: The request to the Slack API failed. (url: https://www.slack.com/api/views.open)   
The server responded with: {'ok': False, 'error': 'internal_error'}

**Comment by zachmetcalf** [comment]
Is there some bot scope that I'm missing? The event is getting triggered, and I'm able to do lots of other things (e.g. posting messages, reacting, etc.), but it just fails on opening the view modal

**Comment by seratch** [comment]
@zachmetcalf I am not 100% confident but you may be using a different token for the SLACK_BOT_TOKEN. Can you double-check if the SLACK_BOT_TOKEN is issued for the app and the workspace you're working now? You can use https://api.slack.com/methods/auth.test/test to see the metadata associated with the token.

**Comment by zachmetcalf** [comment]
Thanks for the tip! I did end up checking the SLACK_BOT_TOKEN's metadata and everything seemed ok.

But, I ended up just pulling some drastic measures: I ended up re-creating my entire slack app with all of the same settings. After that, everything worked (using the new tokens).

It seems like my original token must have gotten into a bad state somehow. I'll keep an eye out for if this ever happens again, but for now I will close
